### PR TITLE
Release 2.1.0

### PR DESCRIPTION
**Pull request by btseytlin**
* Always use staging lightwood when running CI tests
* Use lightwood ShortTextEncoder for short text
* Interface for validating trained predictors
* Sample for analysis only if a dataframe has over 10k rows
* Transaction is accessible as `Predictor.transaction` after running `.learn` 
* DataSource names
* Single words are always categorical, not short text
* Things that have few unique values are also always categorical
* Fixed bug where unit tests couldn't be run without installing optional dependencies

### CODE REVIEW

**Review by btseytlin — COMMENTED** [review]
Reviewed, all seems to be good

**Review by StpMax — APPROVED** [review]

**Review by George3d6 — APPROVED** [review]

**Review by paxcema — APPROVED** [review]

File: mindsdb_native/libs/controllers/predictor.py [hunk 1]
```diff
@@ -23,9 +23,9 @@ def _get_memory_optimizations(df):
 
     mem_usage_ratio = df_memory/total_memory
 
-    sample_for_analysis = True if mem_usage_ratio >= 0.05 else False
-    sample_for_training = True if mem_usage_ratio >= 0.15 else False
-    disable_lightwood_transform_cache = True if mem_usage_ratio >= 0.15 else False
+    sample_for_analysis = True if mem_usage_ratio >= 0.1 or len(df) > pow(10,4) else False
```
Comment (maximlopin) [inline mindsdb_native/libs/controllers/predictor.py:26]: replacing `len(df)` with `df.shape[0] * df.shape[1]` makes more sense

**Review by maximlopin — APPROVED** [review]

File: mindsdb_native/libs/controllers/predictor.py (hunk 1 above)
Comment (George3d6) [reply mindsdb_native/libs/controllers/predictor.py:26]: @maximlopin if you can, please make this change, it should indeed workout better to take into account length + nr columns.

Also, realistically, the training sampling and disabling the cache should probably happen lightwood-side.

Comment (maximlopin) [reply mindsdb_native/libs/controllers/predictor.py:26]: https://github.com/mindsdb/mindsdb_native/pull/84

**Pull request merged**